### PR TITLE
Update nrfdl module logging

### PR DESCRIPTION
**Pull request by nicolai-nordic**

### CODE REVIEW

File: src/utils/logLibVersions.ts [hunk 1]
```diff
@@ -62,6 +62,31 @@ const logVersion = (
...
+    version: ModuleVersion,
+    description: string,
+    name: string
+) => {
+    if (version.moduleName === name) {
+        logger.info(`Using ${description} version: ${describe(version)}`);
+        return version;
+    }
+    // @ts-expect-error -- ModuleVersion type has dependencies now which is a ModuleVersion[]
+    return version.dependencies?.map(dep =>
```
Comment (jonasem) [inline src/utils/logLibVersions.ts:85]: Let's hope there never is a cyclic dependency!

File: src/utils/logLibVersions.ts [hunk 2]
```diff
@@ -62,6 +62,31 @@ const logVersion = (
...
+const searchForAndLogDependency = (
+    version: ModuleVersion,
+    description: string,
+    name: string
+) => {
+    if (version.moduleName === name) {
+        logger.info(`Using ${description} version: ${describe(version)}`);
+        return version;
+    }
+    // @ts-expect-error -- ModuleVersion type has dependencies now which is a ModuleVersion[]
```
Comment (jonasem) [inline src/utils/logLibVersions.ts:84]: Should probably be fixed in nrfdl-js

File: src/utils/logLibVersions.ts [hunk 3]
```diff
@@ -62,6 +62,31 @@ const logVersion = (
...
+};
+
+const searchForAndLogDependency = (
+    version: ModuleVersion,
+    description: string,
+    name: string
+) => {
+    if (version.moduleName === name) {
+        logger.info(`Using ${description} version: ${describe(version)}`);
+        return version;
```
Comment (jonasem) [inline src/utils/logLibVersions.ts:82]: What is this returning to?

**Comment by datenreisender** [comment]
Since master already contains changes (e.g. dependency updates) which we probably do not want to get into the upcoming launcher release, I suggest that you create a branch `release/5.16` off of the tag `v5.16.1` and change this PR to be merged into that branch and then also create the release/tag `v5.16.2` on that branch. 

Then merge the changes from `release/5.16` also into master.

**Pull request closed without merging**